backend/src/controllers/groups-controller.ts

```diff
@@ -77,6 +84,21 @@ export const updateName = async (req: Request, res: Response) =>{
     if(existingGroup){
         return res.status(409).json({ error: "This group name is already taken"});
     }
+    //Checks if UserId exists
+    if(!req.token?.userId){
+        return res.status(422).json({error: "UserId does not exist"});
+    }
+
+    const groupOwner = req.body.userId;
```

Comment:
User shouldn't pass `userId` in the body of the request. E.x. they could pass another user's ID instead of their own. Instead, use the `userId` from the `token`